compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -716,8 +737,29 @@ impl Context {
 
                 let mut dummy_predicate_index = predicate_index;
                 // We must setup the dummy value to match the type of the value we wish to store
-                let dummy =
-                    self.array_get_value(&store_type, block_id, &mut dummy_predicate_index)?;
+                let mut slice_sizes = if store_type.contains_slice_element() {
+                    let store_id = dfg.resolve(store);
+
+                    self.compute_slice_sizes(store_id, None, dfg);
+                    if let Some(slice_sizes) = self.slice_sizes.get_mut(&store) {
+                        slice_sizes.clone()
+                    } else {
+                        return Err(InternalError::UnExpected {
+                            expected: "Store value should have slice sizes computed".to_owned(),
+                            found: "Missing key in slice sizes map".to_owned(),
+                            call_stack: self.acir_context.get_call_stack(),
+                        }
+                        .into());
+                    }
```

Comment:
```suggestion
                    self.slice_sizes.get(&store).cloned().ok_or_else(|| {
                        InternalError::UnExpected {
                            expected: "Store value should have slice sizes computed".to_owned(),
                            found: "Missing key in slice sizes map".to_owned(),
                            call_stack: self.acir_context.get_call_stack(),
                        }
                    })?
```

Reply:
The diff got out of date so I applied this suggestion manually.

Reply:
The diff got out of date so I applied this suggestion manually.